dmscripts/compare_dmp_and_dnb_data.py

```diff
@@ -0,0 +1,324 @@
+        for row in range(1, rows):
+            if s[row - 1] == t[col - 1]:
+                cost = 0  # If the characters are the same in both strings in a given position [i,j] then the cost is 0
+            else:
+                # In order to align the results with those of the Python Levenshtein package,
+                # if we choose to calculate the ratio the cost of a substitution is 2.
+                # If we calculate just distance, then the cost of a substitution is 1.
+                if ratio_calc is True:
+                    cost = 2
+                else:
+                    cost = 1
+            distance[row][col] = min(
+                distance[row - 1][col] + 1,  # Cost of deletions
+                distance[row][col - 1] + 1,  # Cost of insertions
+                distance[row - 1][col - 1] + cost  # Cost of substitutions
+            )
+    if ratio_calc is True:
+        # Computation of the Levenshtein Distance Ratio
+        Ratio = ((len(s) + len(t)) - distance[row][col]) / (len(s) + len(t))
+        return Ratio
+    else:
+        # print(distance) # Uncomment to see the matrix showing how the algorithm computes the cost of deletions,
+        # insertions and/or substitutions
+        # This is the minimum number of edits needed to convert string a to string b
+        return f"The strings are {distance[row][col]} edits away"
+
+
+class DNBAPIException(Exception):
+    "Exception class for the D&B API errors, list here: https://docs.dnb.com/direct/2.0/en-US/response-codes"
+    def __init__(self, code, text):
+        self.code = code
+        self.text = text
+
+
+class DNBDirectAPIClient(object):
+    "A client utilising the D&B direct API with methods used by the DMp API."
+
+    def __init__(self, username, password, stage='production'):
+        "initialisation with D&B credentials and optionally DMp stage"
+        self.stage = stage
+        self.data_api_client = None
+        self.duns_number_compared = set()  # faster in operation than a list
+        self.max_dnb_api_calls_per_method = 10000
+        self.root = 'https://direct.dnb.com'
+        response = requests.post(
+            f'{self.root}/Authentication/V2.0/',
+            headers={
+                'x-dnb-user': username,
+                'x-dnb-pwd': password,
+            }
+        )
+        self.dnb_auth_token = response.json()['AuthenticationDetail']['Token']
+
+    def get_dnb_org_data(self, duns_number):
+        "return a dictionary with all the information D&B has on the company with that DUNS number"
+        response = requests.get(
+            f'{self.root}/V6.0/organizations?match=true&MatchTypeText=Basic&DUNSNumber={str(duns_number)}',
+            headers={
+                'Authorization': self.dnb_auth_token,
+            }
+        )
+        if response.json()['MatchResponse']['TransactionResult']['ResultID'] == 'CM000':
+            return response.json()['MatchResponse']['MatchResponseDetail']['MatchCandidate'][0]
```

Comment:
Could we have a comment explaining what this condition represents?

Reply:
> Could we have a comment explaining what this ~condition represents~ `CM000` thing is about?

Comment explaining it and reference to the D&B documentation.